src/Architecture.cpp

```diff
@@ -186,8 +194,9 @@ Architecture::Architecture(const std::uint16_t nQ, const CouplingMap& cm,
 
 void Architecture::createDistanceTable() {
   isBidirectional = true;
-  Matrix edgeWeights(nqubits, std::vector<double>(
-                                  nqubits, std::numeric_limits<double>::max()));
+  Matrix edgeWeights(
+      nqubits,
+      std::vector<double>(nqubits, std::numeric_limits<double>::infinity()));
```

Comment:
I think in those cases, infinities are really more appropriate (at least from a theoretical standpoint). E.g. the fidelity cost of an edge with error rate 1 is -log(0), which actually tends to infinity. And arguably, the edge cost of 2 qubits that are not connected is not just very high, but infinite.
Are there any practical downsides of using infinities over maximum values here?

Reply:
The "problem" here is buried a bit deeper in mqt-core actually (https://github.com/cda-tum/mqt-core/blob/778bd86f5d0ac9e8cfa00988056e1e8bf038274e/cmake/CompilerOptions.cmake#L43-L45).
We employ `-ffinite-math-only` as a compiler flag in release mode, which is described as
> Allow optimizations for floating-point arithmetic that assume that arguments and results are not NaNs or +-Infs.

I see your point though. Mathematically the infinities make more sense. Although I guess leaving both the infinities and the compiler flag in at the same time is asking for trouble.
Although I am a little hesitant, I would be fine with removing the compiler flag from mqt-core if you believe that is the better option.

Reply:
Oh, I was not aware of that. Thank you for the explanation!

In that case, I'm more in favor of max values, actually. Performance should definitely have precedence in this case. Therefore, dropping an optimization flag, only because the code would be more intuitive or in line with theory, does not make sense in my opinion.